mapclientplugins/geometricfitstep/view/geometricfitwidget.py

```diff
@@ -528,6 +544,115 @@ def _configProjectionCentreGroupsClicked(self):
+            if inheritable:
+                checkBoxTristate = True
+                if isLocallySet is not None:
+                    checkBoxState = QtCore.Qt.PartiallyChecked
+            if isLocallySet:
+                checkBoxState = QtCore.Qt.Checked
+                lineEditDisable = False
+        self._ui.configDataProportion_checkBox.setDisabled(checkBoxDisable)
+        self._ui.configDataProportion_checkBox.setCheckState(checkBoxState)
+        self._ui.configDataProportion_checkBox.setTristate(checkBoxTristate)
+        self._ui.configDataProportion_lineEdit.setDisabled(lineEditDisable)
+        self._ui.configDataProportion_lineEdit.setText(dataProportionStr)
+
+    def _configDataProportionClicked(self):
+        checkState = self._ui.configDataProportion_checkBox.checkState()
+        group = self._ui.configSettingGroup_fieldChooser.getField().getName()
+        self._ui.configDataProportion_lineEdit.setDisabled(True)
+        if checkState == QtCore.Qt.Unchecked:
+            self._getConfig().setGroupDataProportion(group, None)
+        elif checkState == QtCore.Qt.PartiallyChecked:
+            self._getConfig().clearGroupDataProportion(group)
+        else:
+            self._ui.configDataProportion_lineEdit.setDisabled(False)
+            self._configDataProportionEntered()
+        self._updateDataProportion()
+
+    def _configDataProportionEntered(self):
+        value = QLineEdit_parseRealNonNegative(self._ui.configDataProportion_lineEdit)
+        group = self._ui.configSettingGroup_fieldChooser.getField().getName()
+        if value > 0.0:
+            self._getConfig().setGroupDataProportion(group, value)
+        else:
+            print("Invalid model Data Proportion entered")
+        self._updateConfigWidgets()
+
+    def _updateDataWeight(self):
+        realFormat = "{:.4g}"
+        lineEditDisable = True
+        checkBoxDisable = True
+        checkBoxTristate = False
+        dataWeightStr = ""
+        checkBoxState = QtCore.Qt.Unchecked
+        config = self._getConfig()
+        group = self._ui.configSettingGroup_fieldChooser.getField()
+        if group:
+            checkBoxDisable = False
+            groupName = group.getName()
+            dataWeight, isLocallySet, inheritable = config.getGroupDataWeight(groupName)
+            dataWeightStr = realFormat.format(dataWeight)
+            #print("group:", group.getName(), "Weight", dataWeight, isLocallySet,inheritable)
+            if inheritable:
+                checkBoxTristate = True
+                if isLocallySet is not None:
+                    checkBoxState = QtCore.Qt.PartiallyChecked
+            if isLocallySet:
+                checkBoxState = QtCore.Qt.Checked
+                lineEditDisable = False
+        self._ui.configDataWeight_checkBox.setDisabled(checkBoxDisable)
+        self._ui.configDataWeight_checkBox.setCheckState(checkBoxState)
+        self._ui.configDataWeight_checkBox.setTristate(checkBoxTristate)
```

Comment:
Should call setTristate before setCheckState, so PartiallyChecked is a valid state at the time it is set.